Agents/DistrictHeatingOptimisationAgent/README.md

```diff
@@ -0,0 +1,181 @@
+KG_PASSWORD
+FILE_SERVER_ENDPOINT
+FILE_SERVER_USERNAME
+FILE_SERVER_PASSWORD
+```
+
+The `STACK_NAME` variable is used to identify the deployment mode of the agent: In case the `STACK_NAME` is left blank, Postgres and Blazegraph endpoint settings will be taken from the `docker-compose file` values. Otherwise they will be retrieved using the StackClients based on the provided `NAMESPACE` and `DATABASE` variables.
+
+Please note: 
+- When deployed via the Stack manager, the `STACK_NAME` environment variable is set automatically and can be omitted in the config file
+- `SPARQL_QUERY_ENDPOINT` and `SPARQL_UPDATE_ENDPOINT` are both required inputs even for stack deployment (due to default derivation agent config); however, as the actual values for initialising the DH Optimisation Agent are retrieved via the Stack Clients, they can be left blank.
+
+## 1.2 Miscellaneous
+
+**Only relevant** if you intend to build (and publish) the Docker image: Ensure access to Github container registry, as
+- the required `stack-clients-*.jar` resource to be added to [py4jps] during building the Docker image is retrieved from the published Stack-Clients docker image
+- a `publish_docker_image.sh` convenience script is provided to build and publish the agent image to the [Github container registry]. To publish a new image, your github user name and [personal access token] (which must have a `scope` that [allows you to publish and install packages]) needs to be provided. 
+
+## 1.3 Required Derivation Markup
+
+Before any optimisation can be created (and instantiated), all required inputs need to be properly instantiated. This includes:
+- 1 `time:Interval`: the interval to optimise (i.e., defining start and end time of the optimisation to be created, both bounds inclusive)
+- 5 `ts:Forecast`s : time series dataIRIs for the forecasted `ohn:HeatDemand` and 4 `om:Temperature`s, denoting flow and return temperatures at the EfW and municipal heating plant, respectively (the mapping between the provided forecast IRIs and their respective meaning is done internally inside the agent based on associated rdf type and triple patterns)
+
+Used namespaces:
+```xml
+time  : http://www.w3.org/2006/time#
+ts    : https://www.theworldavatar.com/kg/ontotimeseries/
+om    : http://www.ontology-of-units-of-measure.org/resource/om-2/
+ohn   : https://www.theworldavatar.com/kg/ontoheatnetwork/
+```
+
+&nbsp;
+# 2. Agent Operation
+
+The agent is implemented as [derivation agent] using `DerivationWithTimeSeries`. Please note: 1) derivations with time series are currently restricted to synchronous derivations, i.e., derivations which get computed immediately upon request and 2) derivations with time series do not return any specific output triples, as all updates to the time series are expected to be conducted within the agent logic. Initial derivation output triples will only be generated when creating new derivation using `createSyncDerivationForNewInfo`.
+
+## 2.1 Generation Optimisation
```

Comment:
Section 2.2 is quite clear on how to trigger an evaluation, which is nice. By contrast, there is no word in section 2.1 on how to trigger a generation optimisation. Please add that.

Reply:
Addressed by [b6d9c3c](https://github.com/cambridge-cares/TheWorldAvatar/pull/1031/commits/b6d9c3cf9df88d31bedcdff24afb653224a127c0)